### PR TITLE
improvement publishr

### PR DESCRIPTION
**Pull request by boygirl**
cc/ @ryan-roemer does this look right to you?

### CODE REVIEW

**Comment by ryan-roemer** [comment]
@boygirl -- I think you also need something like:

``` js
    "publishr": {
      "dependencies": ["^builder"]
    }
```

@jmcriffey -- Can we get your help / guidance / review on this one? First target repo is: https://github.com/FormidableLabs/victory-core/blob/master/package.json

I think we also may need something analogous to https://github.com/FormidableLabs/publishr/blob/master/.npmignore.publishr too (which will need to live in this archetype). Maybe something like adding:

``` js
    "publishr": {
      "dependencies": ["^builder"]
      "files": {
        ".npmignore": "node_modules/builder-victory-component/config/OTHER_DIR?/npmignore.publishr",
        ".someconfig": ".someconfig.publishr"
      }
    }
```

Target for  `victory-*` repo is:

```
lib/
dist/
demo/ # Maybe? @boygirl?
```

But within the above, we need to exclude:

```
dist/*.map
```

Also, 

Thanks!

**Comment by boygirl** [comment]
@ryan-roemer I think the `publishr` dependency swapping needs to go in the individual repos, since it's acting on their `package.json`. Am I missing something?

**Comment by ryan-roemer** [comment]
@boygirl -- Good point. Remove here and target `victory-*/package.json:devDependencies`.

**Comment by boygirl** [comment]
@ryan-roemer `.npmignore` in `victory-*` is already _not_ ignoring `dist` and `lib`, and is ignoring `dist/*.map`. I think I can skip the `.npmignore.publishr` swapping in this case?

**Comment by ryan-roemer** [comment]
@boygirl -- That sounds right. 👍

File: package.json [hunk 1]
```diff
@@ -16,6 +16,8 @@
     "npm:postinstall": "cd lib || builder run build --expand-archetype",
     "npm:preversion": "builder run check",
     "npm:version": "builder run clean && builder run build",
+    "npm:postversion": "publishr postversion",
+    "npm:postpublish": "publishr postpublish && builder run clean",
```
Comment (jeffrifwald) [inline package.json]: This looks good. You might also want to include a command to contain `publishr dry-run` so you can catch errors before actually versioning or publishing.

**Comment by boygirl** [comment]
Alright, I've added a dry-run script and removed `builder run clean` from `postpublish`. This is unnecessary now as `lib/` and `dist/` will be `.gitignore`'d

@ryan-roemer @jmcriffey good to merge?

**Comment by jeffrifwald** [comment]
LGTM

**Pull request merged**